README.md

```diff
@@ -1,82 +1,69 @@
 
-### Set the swarm to listen for incoming streams
+And use it on your Node.js code as:
 
-```javascript
-s.listen([port], [callback]) // `port` defaults to 4001, `callback` gets called when the socket starts listening
-```
-
-### Close the listener/socket and every open stream that was multiplexed on it
+```JavaScript
+var Swarm = require('libp2p-swarm')
 
-```javascript
-s.closeListener()
+var sw = new Swarm(peerInfoSelf)
 ```
 
-### Register a protocol to be handled by an incoming stream
-
-```javascript
-s.registerHandler('/name/protocol/you/want/version', function (stream) {})
-```
+peerInfoSelf is a PeerInfo object that represents the peer creating this swarm instance.
 
-### Open a new connection
+### Support a transport
 
-Used when we want to make sure we can connect to a given peer, but do not intend to establish a stream with any of the services offered right away.
+libp2p-swarm expects transports that implement [abstract-transport](https://github.com/diasdavid/abstract-transport). For example [libp2p-tcp](https://github.com/diasdavid/node-libp2p-tcp), a simple shim on top of the `net` module to make it work with swarm expectations.
 
-```
-s.openConnection(peerConnection, function (err) {})
+```JavaScript
+sw.addTransport(transport, [options, dialOptions, listenOptions])
 ```
 
+### Add an connection upgrade
 
-### Dial a new stream
+A connection upgrade must be able to receive and return something that implements the [abstract-connection]() interface.
 
+```JavaScript
+sw.addUpgrade(connUpgrade, [options])
 ```
-s.openStream(peerInfo, protocol, function (err, stream) {})
-```
-
-peerInfo must be a [`ipfs-peer`](https://www.npmjs.com/package/ipfs-peer) object, contaning both peer-id and multiaddrs.
 
-## Events emitted
+Upgrading a connection to use a Stream Muxer is still considered a upgrade, but a special case since once this connection is applied, the returned obj will implement the [abstract-stream-muxer]() interface.
 
+```JavaScript
+sw.addStreamMuxer(streamMuxer, [options])
 ```
-.on('error')
 
-.on('connection')
-.on('connection-unknown') // used by Identify to start the Identify protocol from listener to dialer
-```
-
-## Identify protocol
-
-The Identify protocol is an integral part to Swarm. It enables peers to share observedAddrs, identities and other possible address available. This enables us to do better NAT traversal.
-
-To instantiate Identify:
+### Dial to another peer
 
+```JavaScript
+sw.dial(PeerInfo, options, protocol)
+sw.dial(PeerInfo, options) 
 ```
-var Identify = require('ipfs-swarm/identify')
 
-var i = new Identify(swarmInstance, peerSelf)
-```
+dial uses the best transport (whatever works first, in the future we can have some criteria), and jump starts the connection until the point we have to negotiate the protocol. If a Muxer is available, then drop the muxer onto that connection. Good to warm up connections or to check for connectivity. If we have already a Muxer for that peerInfo, than do nothing.
```

Comment:
Because my brain was seeing it as a constructor and constructors are capitalised as a good practice in code, but yep, not in text. Thanks! :)